File: src/django_app/tables/import_export/serializers/llm_model.py
from rest_framework import serializers

from tables.models import LLMModel, Provider


class LLMModelImportSerializer(serializers.ModelSerializer):
    provider_id = serializers.PrimaryKeyRelatedField(
        queryset=Provider.objects.all(),
        source="llm_provider",
        write_only=True,
    )

    class Meta:
        model = LLMModel
        exclude = ["llm_provider"]

    def to_representation(self, instance):
        ret = super().to_representation(instance)
        ret["provider_name"] = instance.llm_provider.name
        return ret

File: frontend/src/app/shared/models/llm.model.ts
export interface LLM_Model {
  id: number;
  name: string;
  description: string | null;
  base_url: string | null;
  deployment: string | null;

  llm_provider: number;
  providerName?: string;
}

export interface GetLlmModelRequest {
  id: number;
  name: string;
  description: string | null;
  base_url: string | null;
  deployment: string | null;

  llm_provider: number;
}
